src/DetailsPage/Documents/FilesUpload.js

```diff
@@ -93,30 +97,35 @@ class FileUpload extends Component {
           };
 
           // Determine total number of files. Return once all files are processed
-          this.setState({ metadata: [...this.state.metadata, fileMetadata] }, () => {
-            const totalFiles = this.pond.getFiles().length;
-            if (this.state.metadata.length === totalFiles) {
-              uploadComplete(this.state.metadata);
-            }
+          setMetadata(prevMetaData => {
+            return ([...prevMetaData, fileMetadata])
           });
         }
       );
     });
   };
 
-  render() {
-    return (
-      <div>
-        <FilePond
-          ref={ref => (this.pond = ref)}
-          allowMultiple
-          maxFiles={5}
-          server={{ process: this.handleProcess }}
-        />
-        <Notification />
-      </div>
-    );
-  }
+  useEffect(() => {
+    if(!isUploading) return;
+    const totalFiles = pond.current.getFiles().length;
+    console.log(metadata.length, totalFiles)
+    if (metadata.length === totalFiles) {
+      uploadComplete(metadata);
+      setIsUploading(false);
+    }
+  }, [isUploading, metadata]);
```

Comment:
You don't need to trigger the `uploadComplete()` function when the `isUploading` flag changes, as the metadata is still empty. 
This is the first issue with document uploading